include/swift/SIL/SILFunction.h

```diff
@@ -879,11 +875,20 @@ class SILFunction
 
   /// Returns true if this function can be inlined into a fragile function
   /// body.
-  bool hasValidLinkageForFragileInline() const { return isSerialized(); }
+  /// If the caller is [serialized] due to explicit `@inlinable` but
+  /// the callee was [serialized_for_package] due to package-cmo, the
+  /// callee is [serialized] and might contain loadable types, which
+  /// is normally disallowed. This check prevents such loadable types
+  /// from being inlined.
+  bool canBeSerializedIntoCaller(
+      std::optional<SerializedKind_t> callerSerializedKind =
+                                 std::nullopt) const;
 
   /// Returns true if this function can be referenced from a fragile function
   /// body.
-  bool hasValidLinkageForFragileRef() const;
+  bool hasValidLinkageForFragileRef(
```

Comment:
We should remove the optional here and always pass the referencing entity's SerializationKind_t.

It can come from the enclosing/referencing abstraction: function, v-table, witness table,  property, movedeinit.